.github/workflows/release.yml

```diff
@@ -14,40 +14,103 @@ env:
+
+          # output is: "<CHKSUM>;<$XCFRAME_ZIP_PATH>"
+          OUTPUT_OF_BUILD=$(bash scripts/ios/build-sargon.sh --release-tag "$NEXT_TAG" | tail -n 1) || exit $?
+          if [[ "$OUTPUT_OF_BUILD" == "BUILT_WITHOUT_RELEASE" ]]; then
+              echo "Error, failed to build, did you forget to pass '--release' to build script? Otherwise check if build-sargon.sh has recently been changed (to something incorrect...)"
+              exit 1;
+          fi
+          CHECKSUM=`echo "$OUTPUT_OF_BUILD" | cut -d ";" -f 1` || exit $?
+          XCFRAME_ZIP_PATH=`echo "$OUTPUT_OF_BUILD" | cut -d ";" -f 2` || exit $?
 
-      - name: Run iOS build script
+          echo "🚢  CHECKSUM: $CHECKSUM"
+          echo "🚢  XCFRAME_ZIP_PATH: $XCFRAME_ZIP_PATH"
+
+          echo "🚢 Ensuring Sargon build for release - that it will work for e.g. iOS wallet to archive."
+          sed -i '' 's/let useLocalFramework = false/let useLocalFramework = true/' Package.swift
+          swift build -c release || exit $?
+          echo "🚢 Swift Sargon builds for release ✅"
+
+          echo "XCFRAME_ZIP_PATH=$XCFRAME_ZIP_PATH" >> "$GITHUB_OUTPUT"
+        env:
+          NEXT_TAG: ${{ steps.tags.outputs.NEXT_TAG }}
+
+      - name: Push release tag
+        if: github.ref == 'refs/heads/main'
+        run: |
+          set -euo pipefail
+
+          # We have .gitigored Sargon.swift because we dont need it in git history, but we
+          # need it for this release, so we must FORCE add it (since it is ignored).
+          echo "🚢  Staging changed files"
+          git add --force Package.swift apple/Sources/UniFFI/Sargon.swift
+
+          git config user.name 'github-actions[bot]'
+          git config user.email 'github-actions[bot]@users.noreply.github.com'
+
+          echo "🚢  Git commiting changes"
+          git status
+          git commit -m "Release of '$NEXT_TAG'" \
+                     -m "Updated Package.swift with new checksum and path to zip on Github, and maybe apple/Sources/UniFFI/Sargon.swift." \
+                     -m "This commit is not merged into main branch (and need not be)." \
+                     --no-verify
+
+          echo "🚢 🏷️ 📡 Pushing tag: $NEXT_TAG, but only tag, not commit."
+          git push origin $NEXT_TAG
         env:
+          NEXT_TAG: ${{ steps.tags.outputs.NEXT_TAG }}
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Create release
+        if: github.ref == 'refs/heads/main'
         run: |
-          rustup target add aarch64-apple-darwin aarch64-apple-ios aarch64-apple-ios-sim
-          sh ./scripts/ios/release.sh
+          # This MUST match whatever you we have declared in Package.swift
+          SWIFT_SARGON_BINARY_ASSET_NAME="libsargon-rs.xcframework.zip"
+
+          GH_RELEASE_TITLE="v$NEXT_TAG"
+          gh release create $NEXT_TAG '$XCFRAME_ZIP_PATH#$SWIFT_SARGON_BINARY_ASSET_NAME' --generate-notes --notes-start-tag $LAST_TAG --title '$GH_RELEASE_TITLE'
```

Comment:
same as https://github.com/radixdlt/sargon/pull/267/files#r1846424837 this job will only work if the build artifact from job `build` is "shared" with this step.